include/TimestampChecker.h

```diff
@@ -59,30 +77,50 @@ class TimestampChecker
 		// The difference between expected duration as per timestamp and actual duration since last frame
 		auto diff = sameOrder ? GetDiff(durationOnTs, durationOnRecv) : (durationOnTs + durationOnRecv);
 		
+		auto result = CheckResult::Valid;
+		int64_t rectifiedTs = 0;
+		
 		bool valid = diff < maxDurationDiffMs;
 		if (valid)
 		{
 			lastTimeStamp = timestamp;
 			lastRecvTime = recvTimeMs;
 			
 			numContinousInvalidFrames = 0;
+			
+			rectifiedTs = timestamp + offset;
+			
+			result = CheckResult::Valid;
 		}
 		else
 		{
 			numContinousInvalidFrames++;
-		}
-		
-		// If there are continous invalid frames, it could be due to reference wrong. Reset to current.
-		if (numContinousInvalidFrames > maxContinousInvalidFrames)
-		{
-			lastTimeStamp = timestamp;
-			lastRecvTime = recvTimeMs;
-			numContinousInvalidFrames = 0;
 			
-			return CheckResult::Reset;
-		}
+			rectifiedTs = (int64_t(recvTimeMs) - int64_t(lastRecvTime))* clock / 1000 + lastTimeStamp + offset;
 
-		return valid ? CheckResult::Valid : CheckResult::Invalid;
+			// If there are continous invalid frames, it could be due to reference wrong. Reset to current.
+			if (numContinousInvalidFrames > maxContinousInvalidFrames)
+			{
+				lastTimeStamp = timestamp;
+				lastRecvTime = recvTimeMs;
+				numContinousInvalidFrames = 0;
+				
+				offset = rectifiedTs - timestamp;
+				
+				result = CheckResult::Reset;
+			}
+			else
+			{
+				result = CheckResult::Invalid;
+			}
+		}
+		
+		return { result, uint64_t(rectifiedTs) };
+	}
+	
+	inline int64_t GetTimestampOffset() const
+	{
+		return offset;
 	}
 	
 	inline static constexpr uint64_t GetDiff(uint64_t lhs, uint64_t rhs)
```

Comment:
When reading the code It is a bit confusing that the dif is the absolute diff:
```suggestion
	inline static constexpr uint64_t GetAbsDiff(uint64_t lhs, uint64_t rhs)
```

Reply:
Done.